src/SprykerEco/Zed/PunchoutCatalogs/Business/PunchoutCatalogsFacade.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/**
+ * Copyright © 2016-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerEco\Zed\PunchoutCatalogs\Business;
+
+use Generated\Shared\Transfer\PunchoutCatalogConnectionTransfer;
+use Generated\Shared\Transfer\PunchoutCatalogResponseTransfer;
+use Generated\Shared\Transfer\PunchoutCatalogTransactionTransfer;
+use Spryker\Zed\Kernel\Business\AbstractFacade;
+
+/**
+ * @method \SprykerEco\Zed\PunchoutCatalogs\Business\PunchoutCatalogsBusinessFactory getFactory()
+ * @method \SprykerEco\Zed\PunchoutCatalogs\Persistence\PunchoutCatalogsEntityManagerInterface getEntityManager()
+ * @method \SprykerEco\Zed\PunchoutCatalogs\Persistence\PunchoutCatalogsRepositoryInterface getRepository()
+ */
+class PunchoutCatalogsFacade extends AbstractFacade implements PunchoutCatalogsFacadeInterface
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @api
+     *
+     * @param int $idConnection
+     *
+     * @return \Generated\Shared\Transfer\PunchoutCatalogConnectionTransfer|null
+     */
+    public function findConnectionById(int $idConnection): ?PunchoutCatalogConnectionTransfer
```

Comment:
@demkos I would recommend to explode this method and have a separate `getPasswordById`. The current method has 4 calls but i'm not sure if everyone uses the password field from it. I would try to reduce
- the access to Vault
- the transferring of password across communication
when it's not necessary

What do you think?

Reply:
After using `findConnectionById` we are using `updateConnection`.
In case of having a separate method with/without loading password, we will need to change `updateConnection` to make sure that empty password will replace the actual password.

Reply:
Created separate method `findConnectionByIdWithPassword`